blob/service.go

```diff
@@ -198,8 +212,12 @@ func (s *Service) getByCommitment(
 
 	namespacedShares, err := s.shareGetter.GetSharesByNamespace(ctx, header.DAH, nID)
 	if err != nil {
-		return nil, nil, err
+		if strings.Contains(err.Error(), share.ErrNotFound.Error()) {
+			err = ErrBlobNotFound
+		}
+		return nil, nil, ErrBlobNotFound
```

Comment:
also if you're returning `ErrBlobNotFound` anyway, no need for this additional check

Reply:
Oooops, it should be an err